thirdeye/thirdeye-frontend/app/pods/components/rootcause-chart-toolbar/component.js

```diff
@@ -42,12 +65,31 @@ export default Component.extend({
 
   granularity: null, // ""
 
+  /**
+   * Selected value in the granularity dropdown
+   * @type {String}
+   */
+  selectedOption: computed('granularity', function() {
+    const keys = Object.keys(this.granularityMapping);
+    let selected = '';
+
+    // Find the key in granularityMapping whose value matches the granularity
+    keys.some(key => {
+      const value = this.granularityMapping[key];
+      if (value === get(this, 'granularity')) {
+        selected = key;
+      }
+    });
+
```

Comment:
@justYves Yeah I was itching about that. Should I extract the translation logic here into a util?

Reply:
@ttbach imo the "correct" way would be to make this a two-sided map

Reply:
I agree. The only thing is there's no bi-directional map in javascript 😞

Reply:
how about this?
`
const granularityValue = get(this, 'granularity');
const selected = keys.find(key => granularityValue === this.granularityMapping[key])
`

Reply:
Ooh i like that. Thanks for the suggestion! @igbopie.